public/scripts/experiments/estimation/estimation.js

```diff
@@ -0,0 +1,465 @@
+import {initialize_random_order} from "/scripts/experiment-properties/balancing/random_generator.js";
+
+import {get_data, get_data_subset} from "/scripts/experiment-properties/data/data_controller.js";
+import {prepare_coordinates,
+    randomize_position,
+    randomize_radius_position,
+    force_greater_right_position} from "/scripts/helpers/experiment_helpers.js";
+
+export default class Estimation {
+    /**
+     * Initializes a Estimation experiment object.
+     *
+     * @param  params          {object}    Parameters passed in from routing
+     */
+    constructor(params) {
+        // Validate fields of params
+        if (params.condition !== 'shape_estimation') {
+            throw  Error("unexpected condition name " + params.condition);
+        }
+        this.condition_name = params.condition;
+        if (params.range !== "estimation") {
+            throw  Error("unexpected range " + params.range);
+        }
+        this.range = params.range;
+        if (params.graph_type !== "shapes") {
+            throw Error("graph type: " + params.graph_type + " is not supported.")}
+        else {
+            this.graph_type = params.graph_type;
+        }
+        if (params.balancing !== "random") {
+            throw Error("balancing: "+ params.balancing + " is not supported.") }
+        else {
+            this.balancing = params.balancing;
+        }
+        this.subject_id = params["subject_id"];
+        this.subject_initials = params["subject_initials"];
+
+        // ========================================
+        // EXPERIMENT CONSTANTS
+        this.MAX_STEP_INTERVAL = 10;
+        this.ROUNDS_PER_COND = 4;
+        this.MAX_Y_POS_JITTER = 0.1; // y axis can be shifted away from default (window / 2) by at most 0.1 * ImageHeight;
+        this.MAX_STEP_SIZE = 0.05; // how much can the size of shapes can be changed at one keypress
+
+        // 1cm is 37.7952755906 pixels
+        this.PIXEL_TO_CM = 37.7952755906;
```

Comment:
Might have to do this dynamically since I think they're going to run this on the lab monitors. I did it by passing in a conversion factor for jnd_radius - looking at jnd_radius_experiment.html has the computation for it.